low_level_platform/impl/src/lf_rp2040_support.c

```diff
@@ -202,9 +201,125 @@ int lf_enable_interrupts_nested() {
  */
 int _lf_single_threaded_notify_of_event() {
   // notify main sleep loop of event
-  sem_release(&_lf_sem_irq_event);
+  if (sem_release(&_lf_sem_irq_event)) {
+      return 0;
+  }
+  return 1;
+}
+
+#else // LF_SINGLE_THREADED
+
+#warning "Threaded runtime on RP2040 is still experimental"
+
+/**
+ * @brief Get the number of cores on the host machine.
+ */
+int lf_available_cores() {
+  return 2;
+}
+
+static void *(*thread_1) (void *);
+static void* thread_1_args;
+static int num_create_threads_called = 0;
+static semaphore_t thread_1_done;
+static void* thread_1_return;
+
+#define MAGIC_THREAD1_ID 314159
+
+void core1_entry() {
+  thread_1_return = thread_1(thread_1_args);
+  sem_reset(&thread_1_done, 1);
+}
+
+int lf_thread_create(lf_thread_t* thread, void *(*lf_thread) (void *), void* arguments) {
```

Comment:
It seems like this could be easily generalized to support NUM_CORES rather than just 2.  It's not essential to do that now, so if you don't have time, just mark this resolved and open an issue.  We would need a platform to test it on anyway.

Reply:
Yup, that would be a good thing to do. This would be very relevant to an stm32 platform if one were added in the future. I'll open an issue for it